tests/plugins/clipboard/_helpers/pasting.js

```diff
@@ -3,6 +3,48 @@ getDefaultNotification, createFixtures */
 
 'use strict';
 
+// Mock FileReader.
+( function() {
+	var fileMockBase64 = ';base64,fileMockBase64=',
+		fileMockType,
+		readResultMock;
+
+	function FileReaderMock() {
+		this.listeners = {};
+	}
+
+	// Any MIME type.
+	FileReaderMock.setFileMockType = function( type ) {
+		fileMockType = type;
+	};
+
+	// Result can be: load, abort, error.
+	FileReaderMock.setReadResult = function( readResult ) {
+		readResultMock = readResult;
+		if ( !readResultMock ) {
+			readResultMock = 'load';
+		}
+	};
+
+	FileReaderMock.prototype.addEventListener = function( eventName, callback ) {
+		this.listeners[ eventName ] = callback;
+	};
+
+	FileReaderMock.prototype.readAsDataURL = function() {
+		CKEDITOR.tools.setTimeout( function() {
+			this.result = ( readResultMock == 'load' ? 'data:' + fileMockType + fileMockBase64 : null );
+
+			if ( this.listeners[ readResultMock ] ) {
+				this.listeners[ readResultMock ]();
+			}
+		}, 15, this );
+	};
+
+	/* jshint ignore:start */
+	FileReader = FileReaderMock;
+	/* jshint ignore:end */
```

Comment:
I'm wondering if

```suggestion
	window.FileReader = FileReaderMock;
```

would still need jshint comments.